mail_activity_done/models/mail_activity.py

```diff
@@ -9,16 +9,14 @@ class MailActivity(models.Model):
 
     active = fields.Boolean(default=True)
     done = fields.Boolean(default=False)
-    state = fields.Selection(selection_add=[("done", "Done")], compute="_compute_state")
-    date_done = fields.Date(
-        "Completed Date",
-        index=True,
-        readonly=True,
+    state = fields.Selection(
+        selection_add=[("done", "Done")], compute="_compute_state", store=True
```

Comment:
Why did you make the field stored?

Reply:
Hi @CasVissers-360ERP I made it stored as you get an error with filter "Completed Activities" in mail_activity_view_search  when field state is not stored. Any suggestion? Thank you.

Reply:
Nop! Only wondering why this was not an issue in the v13 version?

Reply:
@manuelregidor In addition this change breaks the standard method of Odoo since the state is based on the date of the activity. When you open the list view you want the state to be computed based on the current date, the current depends don't solve this.

Reply:
@CasVissers-360ERP was looking on why it was not an issue in 13.0 The thing is validation was changed in 14.0:
postprocess method in ir_ui_view calls _validate_attrs which wasn't there in 13.0 and has the following code;

https://github.com/odoo/odoo/blob/63500d6357f6f27724d1cda44db071aba968515a/odoo/addons/base/models/ir_ui_view.py#L1292-L1294

That _get_server_domain_variables is the culprit as it states:
https://github.com/odoo/odoo/blob/63500d6357f6f27724d1cda44db071aba968515a/odoo/addons/base/models/ir_ui_view.py#L1528-L1532

And the definition of that `_description_searchable` is:
https://github.com/odoo/odoo/blob/63500d6357f6f27724d1cda44db071aba968515a/odoo/fields.py#L687-L689

So that means for this to be working without the state field being stored we need to provide search method.

Reply:
Update: adding search method on the `state` fields makes it work. At least it seems so. And the method doesn;t even seems to be called. Ever.

Reply:
```suggestion
        selection_add=[("done", "Done")], compute="_compute_state", search="_search_state"
```